Source/WebKit/Shared/TextExtractionToStringConversion.cpp

```diff
@@ -469,6 +467,27 @@ static void addPartsForItem(const TextExtraction::Item& item, std::optional<Node
     );
 }
 
+static bool childTextNodeIsRedundant(const TextExtraction::Item& parent, const String& childText)
+{
+    if (auto link = parent.dataAs<TextExtraction::LinkItemData>(); link && link->completedURL.string().containsIgnoringASCIICase(childText))
+        return true;
+
+    if (auto formControl = parent.dataAs<TextExtraction::TextFormControlData>()) {
+        auto& editable = formControl->editable;
+        if (editable.placeholder.containsIgnoringASCIICase(childText))
+            return true;
+
+        if (editable.label.containsIgnoringASCIICase(childText))
+            return true;
+
+        return std::any_of(parent.ariaAttributes.begin(), parent.ariaAttributes.end(), [&](auto& entry) {
```

Comment:
nit: I am unsure if `HashMap` works with `std::ranges`, but:
```suggestion
        return std::ranges::any_of(parent.ariaAttributes, [&](auto& entry) {
```

Reply:
Yep, that works and is much cleaner — thanks for the review!